lib/vector-core/src/event/metric/tags.rs

```diff
@@ -455,14 +467,13 @@ mod tests {
             }
         }
 
-        #[test]
-        fn tag_value_set_checks(values: Vec<TagValue>, addition: TagValue) {
+        fn tag_value_nonbare_set_checks(values: Vec<String>, addition: String) {
             let mut set = TagValueSet::from(values.clone());
             assert_eq!(set.is_empty(), values.is_empty());
             // All input values are contained in the set.
-            assert!(values.iter().all(|v| set.contains(v.as_str())));
+            assert!(values.iter().all(|v| set.contains(&Some(v.clone()))));
             // All set values were in the input.
-            assert!(set.iter().all(|v| values.contains(&v.into())));
+            assert!(set.iter().all(|s| s.map_or(true, |s| values.contains(&s.to_string()))));
```

Comment:
In this test case, given that we're always going to get valid `String` values in `values`... should we ever actually see `None` when iterating the tag value set?

Reply:
No, we shouldn't, but I'm not sure what the question is here. This pair of assertions is checking that `values == set` without regard to ordering.

Reply:
I guess the question would be: could this be `values.contains(s.expect("should not be none"))` or something like that?